Documentation/build-arm64-firmware.md

```diff
@@ -0,0 +1,211 @@
+Building and Updating ARM64 Firmware
+==============
+
+This document describes how to set up a build environment to build the latest firmware, update the firmware on the SD Card for testing and development, and include the new firmware in the FFU builds.
+
+## Setting up your build environment
+
+1) Set up a Linux environment.
+    * Dedicated Linux system
+    * Linux Virtual Machine
+    * Windows Subsystem for Linux ([WSL setup instructions](https://docs.microsoft.com/en-us/windows/wsl/install-win10))
+    * Note: We validate with both Ubuntu in WSL and standalone Ubuntu machines.
+
+1) Update and install build tools
+    ```bash
+    $ sudo apt-get update
+    $ sudo apt-get upgrade
+    $ sudo apt-get install build-essential python python-dev python-crypto python-wand device-tree-compiler bison flex swig iasl uuid-dev wget git bc libssl-dev zlib1g-dev python3-pip
+    $ pushd ~
+    $ wget https://releases.linaro.org/components/toolchain/binaries/7.2-2017.11/aarch64-linux-gnu/gcc-linaro-7.2.1-2017.11-x86_64_aarch64-linux-gnu.tar.xz
+    $ tar xf gcc-linaro-7.2.1-2017.11-x86_64_aarch64-linux-gnu.tar.xz
+    $ rm gcc-linaro-7.2.1-2017.11-x86_64_aarch64-linux-gnu.tar.xz
+    $ pip3 install mu_environment
+    $ popd
+    ```
+
+1) Clone all repositories into the same directory as the imx-iotcore repo.
```

Comment:
The direction I think the IMX firmware build system should go is to use submodules for deterministic versioning and a Makefile to automate the build process, like I did for LSDK (https://github.com/ms-iot/lsdk).